jwplatform/client.py

```diff
@@ -1,15 +1,21 @@
 # -*- coding: utf-8 -*-
 import http.client
+import logging
+from http.client import RemoteDisconnected
 import json
+import os
 import urllib.parse
 
 from jwplatform import __version__
 from jwplatform.errors import APIError
 from jwplatform.response import APIResponse, ResourceResponse, ResourcesResponse
+from jwplatform.upload import MultipartUpload, SingleUpload, UploadType, MIN_PART_SIZE, MaxRetriesExceededError, \
+    UPLOAD_BASE_URL, UploadContext, MAX_FILE_SIZE
 
 JWPLATFORM_API_HOST = 'api.jwplayer.com'
 JWPLATFORM_API_PORT = 443
 USER_AGENT = f"jwplatform_client-python/{__version__}"
+RETRY_COUNT = 3
```

Comment:
I'm trying to understand a bit more how we want to treat retries here. I left it out of the other requests since it's not something I see other libraries doing nor is it something I'd really expect as a user. I'd typically want full control over this sort of thing.

I think it does make sense for a complex managed process like multipart uploads: retries are already expected for media parts so why not do them for everything.

With that context and looking at how this is used now (it only seems to be used for part uploads) it seems like this should be named as such `UPLOAD_REQUEST_ATTEMPTS` or something related to uploads.

I did also find it odd that request_with_retry has a default of 3 but doesn't reference this value but maybe that's okay if this is dedicated to uploads while the request_with_retry method is more generalizable.